lib/superagent-mock.js

```diff
@@ -125,23 +97,34 @@ function mock (superagent, config, logger) {
       }
     }
 
-
     if (querystring.length) {
       path += (~path.indexOf('?') ? '&' : '?') + querystring;
     }
 
-    var parser = testUrlForPatterns(this.url);
+    // Attempt to match url against the patterns in fixtures.
+    var parser = null;
+    for (var i = 0, len = config.length; i < len; i++) {
+      if (new RegExp(config[i].pattern, 'g').test(path)){
```

Comment:
Maybe you should test both, `url` and `path`, and log a warning when the path contains a question mark and there is a matcher that only match the `url` ?